### PR TITLE
rewrite updateRepo to collect and throw errors

### PR DESCRIPTION
**Pull request by pfuehrlich-pik**
The mattermost bot we set up a while ago does not message us on erros, because updateRepo would always catch all errors. Now it will catch and collect all errors at first and then throw them before returning if there were any.

### CODE REVIEW

File: R/updateRepo.R [hunk 1]
```diff
@@ -20,138 +20,127 @@
...
-      }
-    })
+  if (!file.exists(file.path(path, "PACKAGES"))) {
+    write_PACKAGES(path, unpacked = TRUE)
   }
-
-  # build new versions of packages including documentation
-  updatePACKAGES <- FALSE
-  availablePackages <- suppressWarnings(available.packages(paste0("file:", path), filters = "duplicates"))
+  availablePackages <- suppressWarnings(available.packages(paste0("file:", path)))
```
Comment (tscheypidi) [inline R/updateRepo.R:30]: why did you remove the `filter`?

**Review by tscheypidi — APPROVED** [review]
looks good

File: R/updateRepo.R (hunk 1 above)
Comment (pfuehrlich-pik) [reply R/updateRepo.R:30]: There were a couple of places where repository states that should never happen were fixed. Having these kind fixes makes it harder to find/debug the underlying problems, so I removed them. I think this is one of these places, with the state the repo on the RSE server is in right now the filter does not make a difference:
```
> a <- available.packages(paste0("file:."), filters = "duplicates")
> b <- available.packages(paste0("file:."))
> waldo::compare(a, b)
✔ No differences
```
